pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go

```diff
@@ -157,6 +194,14 @@ func (s *CloudStackDatacenterConfigSpec) Equal(o *CloudStackDatacenterConfigSpec
 			return false
 		}
 	}
+	if len(s.AvailabilityZones) != len(o.AvailabilityZones) {
+		return false
+	}
+	for i, az := range s.AvailabilityZones {
+		if !az.Equal(&o.AvailabilityZones[i]) {
```

Comment:
This is relying on order
Since we have a name, I would vote to create a map, and then compare equality searching in that map by name.

Reply:
Makes sense. I think I'll have to change the SetDefaults function then, since that currently defaults an AZ name to "Global" when converting from the old schema to the new one. Validating will become more tricky however, since we won't have a way to map from the old input ini file with the [Global] tag to the new AZ's which are named by their zone names. We can deal with that in the subsequent PR